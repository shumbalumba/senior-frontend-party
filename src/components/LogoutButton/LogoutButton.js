// @flow

import * as React from 'react';
import styled from 'styled-components';

import { LogoutIcon } from '../Icons/Icons';

const LogoutWrapper = styled.span`
  position: absolute;
  right: 0;
  top: 50%;
  transform: translate(-50%, -50%);
  svg {
    margin: 0 10px 3px 0;
  }
  &:hover {
    cursor: pointer;
    color: ${props => props.theme.colour.button};
    svg {
      fill: ${props => props.theme.colour.button};
    }
  }
`;

type LogoutButtonProps = {
  onClick: () => void,
};

export const LogoutButton = ({ onClick }: LogoutButtonProps) => (
  <LogoutWrapper onClick={onClick}>
    <LogoutIcon />
    Logout
  </LogoutWrapper>
);
