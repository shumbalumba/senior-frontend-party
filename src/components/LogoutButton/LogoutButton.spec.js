import * as React from 'react';
import { shallow } from 'enzyme';
import { LogoutButton, LogoutWrapper } from './LogoutButton';

describe('<LogoutButton />', () => {
  it('renders', () => {
    const wrapper = shallow(<LogoutButton onClick={jest.fn()} />);
    expect(wrapper.exists()).toBeTruthy();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<LogoutButton onClick={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<LogoutButton onClick={onClick} />);
    const button = wrapper.find(LogoutWrapper);
    button.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
