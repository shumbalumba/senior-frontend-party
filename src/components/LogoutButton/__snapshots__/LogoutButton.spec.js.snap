// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutButton /> matches snapshot 1`] = `
<styled.span
  onClick={[MockFunction]}
>
  <Component />
  Logout
</styled.span>
`;
